task-impls/src/events.rs

```diff
@@ -10,19 +11,22 @@ use hotshot_types::vote::{DAVote, QuorumVote};
 
 use crate::view_sync::ViewSyncPhase;
 
-#[derive(Debug, Clone)]
+#[derive(Eq, Hash, PartialEq, Debug, Clone)]
 pub enum SequencingHotShotEvent<TYPES: NodeType, I: NodeImplementation<TYPES>> {
     Shutdown,
     QuorumProposalRecv(Proposal<QuorumProposalType<TYPES, I>>, TYPES::SignatureKey),
     QuorumVoteRecv(QuorumVote<TYPES, I::Leaf>, TYPES::SignatureKey),
     DAProposalRecv(Proposal<DAProposal<TYPES>>, TYPES::SignatureKey),
     DAVoteRecv(DAVote<TYPES, I::Leaf>, TYPES::SignatureKey),
-    ViewSyncMessage(ViewSyncMessageType<TYPES>),
+    DACertificateRecv(DACertificate<TYPES>),
     QuorumProposalSend(Proposal<QuorumProposalType<TYPES, I>>),
     QuorumVoteSend(QuorumVote<TYPES, I::Leaf>),
     DAProposalSend(Proposal<DAProposal<TYPES>>),
     DAVoteSend(DAVote<TYPES, I::Leaf>),
+    DACertificateSend(DACertificate<TYPES>),
     QCFormed(QuorumCertificate<TYPES, I::Leaf>),
+    /// A view has timeouted out locally or we received/formed a timeout cert
+    /// contains the view we are moving to
     ViewChange(ViewNumber),
     ViewSyncTimeout(ViewNumber, u64, ViewSyncPhase),
     ViewSyncMessageSend(ViewSyncMessageType<TYPES>),
```

Comment:
I'll have these working in my next PR! Already broke them out into ViewSyncVoteSend/Recv and ViewSyncCertificateSend/Recv